dining/views.py

```diff
@@ -0,0 +1,348 @@
+
+            swipe_balances = []
+
+            for balance in balances:
+                swipe_balances.append(balance.swipes)
+
+            spent = 0
+
+            for i, balance in enumerate(swipe_balances):
+                if i == 0:
+                    continue
+                else:
+                    if swipe_balances[i] < swipe_balances[i - 1]:
+                        continue
+                    else:
+                        spent += swipe_balances[i] - swipe_balances[i - 1]
+
+            eastern = timezone("US/Eastern")
+            now = datetime.now().replace(tzinfo=eastern, hour=0, minute=0, second=0, microsecond=0)
+            days_since_start = (now - start).days
+            spend_rate = spent / days_since_start
+
+            days_left = int(balance / spend_rate)
+
+            broke_day = now + timedelta(days=days_left)
+
+            card["predicted-zero-date"] = broke_day.isoformat()
+
+            if broke_day > end:
+                days_to_end = (end - now).days
+                card["balance-at-semester-end"] = round(spend_rate * days_to_end, 0)
+
+        for balance in balances:
+            card["data"].append({"date": balance.created_at.isoformat(), "balance": balance.swipes})
+
+        card["data"] = sorted(card["data"], key=lambda k: k["date"])
+
+        return card
+
+    def get_frequent_locations(self, uid, start, end):
+
+        transactions = DiningTransaction.objects.filter(
+            account=uid, date__gte=start, amount__lt=0
+        ).exclude(description__icontains="meal_plan")
+
+        if len(transactions) < 10:
+            return None
+
+        venue_spends = (
+            transactions.values("description")
+            .annotate(total_spend_venue=Sum("amount"))
+            .order_by("total_spend_venue")
+        )
+
+        card = {
+            "type": "frequent-locations",
+            "data": [],
+        }
+
+        venues = [
+            {
+                "location": map_to_pretty(venue["description"]),
+                "week": 0,
+                "month": 0,
+                "semester": venue["total_spend_venue"] * -1,
+            }
+            for venue in venue_spends
+        ][:5]
+
+        eastern = timezone("US/Eastern")
+        now = datetime.now().replace(tzinfo=eastern, hour=0, minute=0, second=0, microsecond=0)
+        for transaction in transactions:
+
+            days_since_transaction = (now - transaction.date).days
+            venue_index = next(
+                (
+                    index
+                    for (index, d) in enumerate(venues)
+                    if d["location"] == map_to_pretty(transaction.description)
+                ),
+                None,
+            )
+
+            if venue_index is None:
+                continue
+
+            transaction.amount *= -1
+
+            if days_since_transaction <= 7:
+                venues[venue_index]["week"] += transaction.amount
+            elif days_since_transaction <= 30:
```

Comment:
Ditto with `DAYS_PER_MONTH`. Also, do we want to standardize on 30 per month, even when some are less and more? I think it's fine, just want to make sure it's a conscious decision.